app/frontend/src/features/donations/DonationsBox.tsx

```diff
@@ -0,0 +1,474 @@
+      margin: "initial",
+      height: "100%",
+      width: "100%",
+      display: "flex",
+    },
+    "& > .Mui-checked ~.MuiFormControlLabel-label": {
+      border: `2px solid ${theme.palette.primary.main}`,
+      backgroundColor: theme.palette.background.paper,
+    },
+  },
+
+  buttonSecondaryActive: {
+    "&&": {
+      border: `2px solid ${theme.palette.primary.main}`,
+      backgroundColor: theme.palette.background.paper,
+    },
+  },
+
+  buttonMain: {
+    backgroundColor: theme.palette.primary.main,
+    "&&": {
+      borderRadius: "0.5rem",
+      boxShadow: "initial",
+    },
+    "&:hover": {
+      opacity: 0.4,
+      backgroundColor: theme.palette.primary.main,
+    },
+    "& .MuiButton-label": {
+      color: theme.palette.background.paper,
+      transition: `color ${theme.transitions.duration.short} ${theme.transitions.easing.easeInOut}`,
+    },
+  },
+
+  buttonMainText: {
+    color: theme.palette.background.paper,
+    fontWeight: 700,
+    transition: `color ${theme.transitions.duration.short} ${theme.transitions.easing.easeInOut}`,
+  },
+
+  formGroup: {
+    width: "100%",
+  },
+
+  marginY2: {
+    margin: theme.spacing(2, 0),
+  },
+
+  marginBottom2: {
+    marginBottom: theme.spacing(2),
+  },
+
+  inputWrapper: {
+    position: "relative",
+    "&::before": {
+      content: `'${DONATIONSBOX_CURRENCY}'`,
+      position: "absolute",
+      top: "50%",
+      transform: "translateY(-50%)",
+      left: theme.spacing(1),
+      color: theme.palette.grey[600],
+      fontWeight: 700,
+      fontSize: theme.typography.button.fontSize,
+    },
+  },
+
+  inputNumber: {
+    width: "100%",
+    height: "100%",
+    border: `2px solid ${theme.palette.grey[200]}`,
+    paddingLeft: theme.spacing(2),
+    color: theme.palette.grey[600],
+    fontWeight: 700,
+    fontSize: theme.typography.button.fontSize,
+    "&&": {
+      borderRadius: theme.shape.borderRadius * 2,
+      boxShadow: "initial",
+    },
+    "&:hover": {
+      border: `2px solid ${theme.palette.primary.main}`,
+    },
+    "&:focus-visible": {
+      border: `2px solid ${theme.palette.primary.main}`,
+      outline: "none",
+      boxShadow: "none",
+    },
+  },
+
+  inputNumberActive: {
+    "&&": {
+      border: `2px solid ${theme.palette.primary.main}`,
+    },
+  },
+}));
+
+export interface DonationFormData {
+  amount: number;
+  recurring: boolean;
+}
+
+export default function DonationsBoxMixed() {
+  const stripePromise = loadStripe(process.env.REACT_APP_STRIPE_KEY!);
+
+  const classes = useStyles();
+
+  const [success, setSuccess] = useState(false);
+  const [cancelled, setCancelled] = useState(false);
+
+  const location = useLocation();
+
+  const [isAmount, setIsAmount] = useState(false);
+
+  useEffect(() => {
+    const query = new URLSearchParams(location.search);
+    setSuccess(query.get("success") ? true : false);
+    setCancelled(query.get("cancelled") ? true : false);
+  }, [location]);
```

Comment:
Yeah I can't see why this wouldn't work 👍 

Better still, if you pass `useState` an initialiser function instead, it would be lazily evaluated and only do so once. i.e.

```typescript
const location = useLocation();
const query = new URLSearchParams(location.search);
// Use initialiser function as opposed to passing in the ternary expression directly
const [success, setSuccess] = useState(() => query.get("success") ? true : false);
```

But yeah, not sure it makes a huge difference in this case anyway, since I think most of the work is still done on the line where you re-create the `URLSearchParams` and unless it's proven to be slow, I probably wouldn't bother optimising it.